src/vcpkg-test/platform-expression.cpp

```diff
@@ -174,10 +376,194 @@ TEST_CASE ("weird platform-expressions whitespace", "[platform-expression]")
+}
+
+TEST_CASE ("invalid logic expression, unexpected character", "[platform-expression]")
+{
+    auto m_expr = parse_expr("windows arm");
+    CHECK_FALSE(m_expr);
+}
+
+TEST_CASE ("invalid logic expression, use '|' instead of 'or'", "[platform-expression]")
+{
+    auto m_expr = parse_expr("windows or arm");
+    CHECK_FALSE(m_expr);
+}
+
+TEST_CASE ("unexpected character or identifier in logic expression", "[platform-expression]")
+{
+    auto m_expr = parse_expr("windows aND arm");
+    CHECK_FALSE(m_expr);
+    m_expr = parse_expr("windows a&d arm");
+    CHECK_FALSE(m_expr);
+    m_expr = parse_expr("windows oR arm");
+    CHECK_FALSE(m_expr);
+    m_expr = parse_expr("windows o|r arm");
+    CHECK_FALSE(m_expr);
+}
+
+TEST_CASE ("unexpected identifier in logic expression", "[platform-expression]")
+{
+    auto m_expr = parse_expr("windows amd arm");
+    CHECK_FALSE(m_expr);
+    m_expr = parse_expr("windows andsynonym arm");
+    CHECK_FALSE(m_expr);
+}
+
+TEST_CASE ("missing closing )", "[platform-expression]")
+{
+    auto m_expr = parse_expr("(windows & arm | linux");
+    CHECK_FALSE(m_expr);
+    m_expr = parse_expr("( (windows & arm) | (osx & arm64) | linux");
+    CHECK_FALSE(m_expr);
+}
+
+TEST_CASE ("missing or invalid identifier", "[platform-expression]")
+{
+    auto m_expr = parse_expr("!");
+    CHECK_FALSE(m_expr);
+    m_expr = parse_expr("w!ndows");
+    CHECK_FALSE(m_expr);
+}
+
+TEST_CASE ("mixing & and | is not allowed", "[platform-expression]")
 {
     auto m_expr = parse_expr("windows & arm | linux");
     CHECK_FALSE(m_expr);
     m_expr = parse_expr("windows | !arm & linux");
     CHECK_FALSE(m_expr);
 }
+
+TEST_CASE ("invalid expression, no binary operator", "[platform-expression]")
+{
+    auto m_expr = parse_expr("windows linux");
+    CHECK_FALSE(m_expr);
+    m_expr = parse_expr("windows x64");
+    CHECK_FALSE(m_expr);
+    m_expr = parse_expr("!windows x86");
+    CHECK_FALSE(m_expr);
+}
+
+TEST_CASE ("invalid expression, missing binary operand", "[platform-expression]")
+{
+    auto m_expr = parse_expr("windows & ");
+    CHECK_FALSE(m_expr);
+    m_expr = parse_expr(" | arm");
+    CHECK_FALSE(m_expr);
+    m_expr = parse_expr("windows & !arm & ");
+    CHECK_FALSE(m_expr);
+}
+
+TEST_CASE ("invalid identifier", "[platform-expression]")
+{
+    auto m_expr = parse_expr("windows & x^$");
+    CHECK_FALSE(m_expr);
+}
+
+TEST_CASE ("invalid alternate expressions", "[platform-expression]")
+{
+    auto m_expr = parse_expr("windows an%d arm");
+    CHECK_FALSE(m_expr);
+    m_expr = parse_expr("windows aNd arm");
+    CHECK_FALSE(m_expr);
+    m_expr = parse_expr("windows andMORE arm");
+    CHECK_FALSE(m_expr);
+    m_expr = parse_expr("windows and+ arm");
+    CHECK_FALSE(m_expr);
+    m_expr = parse_expr("windows and& arm");
+    CHECK_FALSE(m_expr);
+
+    m_expr = parse_expr("notANY windows");
+    CHECK_FALSE(m_expr);
+    m_expr = parse_expr("not! windows");
```

Comment:
Why is this one invalid? Shouldn't this parse as `!!windows`?

Reply:
That's a good question. At first glance, one would expect `not! windows` to be the same as `!!windows` and evaluate to windows. But the code currently fails to part both of those expressions. It also doesn't appear to be a valid parse according to the EBNF. 
So I think you're right and this should be valid, but we'll need to update the grammar when we do.